### DISCUSSION TITLE
Specfem 2D - GPU

### DISCUSSION THREAD

**Discussion by etotten93** [SPECFEM/specfem3d#1731] [General]
Hi There,

           I hope this message finds you all well. I see that SPECFEM 
           is GPU-enabled for 3D globe and Cartesian - however, I am 
           wondering if there is a GPU-enabled 2D Cartesian version available.

Many thanks,

E

**Comment by williameaton** [comment]
The Fortran90 [Specfem2d](https://github.com/SPECFEM/specfem2d) can run on GPUs. There is also a C++ version under development [here](https://specfem2d-kokkos.readthedocs.io/en/latest/) which uses Kokkos for compatibility with different GPU architectures. I believe it is so far capable of forward and adjoint simulations.

**Reply by etotten93** [reply]
@williameaton thanks for the quick response. In regard to the Fortran90 version, is this GPU ready i.e. are there instructions and implementation examples for/of its stable offloading to GPU that I can have a look at? Thanks again.

**Reply by williameaton** [reply]
For transparency, I don't really work with Specfem2d so hopefully someone else can give you more details. You would need to configure it with CUDA (if using NVIDIA for example) using the ` --with-cuda`, and also change `GPU_MODE` to `.true.` in the `DATA/Par_file` directory before running the mesher & solver. 

@danielpeter is there a specific example/documentation for working with GPUs?

**Reply by etotten93** [reply]
Thanks for the feedback again. I think the pathway described above makes sense - any specific documentation will likely save time getting this compiled! :)

**Comment by danielpeter** [comment]
there is now a short entry in the [SPECFEM2D wiki](https://github.com/SPECFEM/specfem2d/wiki) about the configuration with CUDA:
[Using the GPU version of the code](https://github.com/SPECFEM/specfem2d/wiki/02_getting_started#using-the-gpu-version-of-the-code)

**Reply by etotten93** [reply]
Thank you @danielpeter. I will let you know how I get on in due course.
